src/twelvedata/websocket.py

```diff
@@ -99,7 +99,13 @@ def keep_alive(self):
             self.heartbeat()
 
     def heartbeat(self):
-        self.ws.send('{"action": "heartbeat"}')
+        if not self.ready:
+            return
+
+        try:
+            self.ws.send('{"action": "heartbeat"}')
+        except Exception as e:
+            self.logger.error("Error calling heartbeat method: {}".format(e))
```

Comment:
Thank you for this fix @sgacode ! Question: Shouldn't it try to reconnect (`self_heal()`) after this line?

Reply:
We use keep_alive method and heartbeat API action inside it for avoiding connection breaks by timeouts. This method running in separate thread.
It makes sense to run it only if the connection is ready and working. 

When connection is broken, the method Event Receiver.on_error will be executed. In this method will be executed TD WebSocket.self_heal for reconnection. 
After successful connection keep_alive method will start working and perform its function.
So calling self_heal inside keep_alive is not necessary.